nixos/modules/services/mail/schleuder.nix

```diff
@@ -0,0 +1,100 @@
+{ config, pkgs, lib, ... }:
+let
+  cfg = config.services.schleuder;
+  settingsFormat = pkgs.formats.yaml { };
+  postfixMap = entries: lib.concatStringsSep "\n" (lib.mapAttrsToList (name: value: "${name} ${value}") entries);
+  writePostfixMap = name: entries: pkgs.writeText name (postfixMap entries);
+in
+{
+  options.services.schleuder = {
+    enable = lib.mkEnableOption "Schleuder secure remailer";
+    lists = lib.mkOption {
+      description = "Addresses of lists that should be handled by Schleuder.";
+      type = lib.types.listOf lib.types.str;
+      default = [ ];
+      example = [ "widget-team@example.com" "security@example.com" ];
+    };
+    /* maybe one day....
+      domains = lib.mkOption {
+      description = "Domains for which mail should be handled by Schleuder.";
+      type = lib.types.listOf lib.types.str;
+      default = [];
+      example = ["securelists.example.com"];
+      };
+    */
+    settings = lib.mkOption {
+      description = "Settings for schleuder.yml";
+      type = settingsFormat.type;
+      default = { };
```

Comment:
Probably best to go for a submodule with a `freeformType = format.type` instead, so we can have some sane presets.

I wouldn't want anyone to be inspired by https://0xacab.org/schleuder/schleuder/-/blob/master/etc/schleuder.yml#L34. We can do such much better.

Reply:
That preset wouldn't be visible on search.nixos.org.